haystack/pipelines/config.py

```diff
@@ -111,9 +127,70 @@ def build_component_dependency_graph(
     return graph
 
 
-def _validate_user_input(input: str):
-    if isinstance(input, str) and not VALID_CODE_GEN_INPUT_REGEX.match(input):
-        raise ValueError(f"'{input}' is not a valid config variable name. Use word characters only.")
+def validate_yaml(path: Path):
+    """
+    Validates the given YAML file using the autogenerated JSON schema.
+
+    :param pipeline_config: the configuration to validate
+    :return: None if validation is successful
+    :raise: `PipelineConfigError` in case of issues.
+    """
+    pipeline_config = read_pipeline_config_from_yaml(path)
+    validate_config(pipeline_config=pipeline_config)
+    logging.debug(f"'{path}' contains valid Haystack pipelines.")
+
+
+def validate_config(pipeline_config: Dict) -> None:
+    """
+    Validates the given configuration using the autogenerated JSON schema.
+
+    :param pipeline_config: the configuration to validate
+    :return: None if validation is successful
+    :raise: `PipelineConfigError` in case of issues.
+    """
+    validate_config_strings(pipeline_config)
+
+    with open(JSON_SCHEMAS_PATH / f"haystack-pipeline-unstable.schema.json", "r") as schema_file:
+        schema = json.load(schema_file)
+
+    compatible_versions = [version["const"].replace('"', "") for version in schema["properties"]["version"]["oneOf"]]
+    try:
+        Draft7Validator(schema).validate(instance=pipeline_config)
```

Comment:
Right, good point!

Reply:
On a second though, we might want to postpone this in another (smaller) PR. The fact is that in any case, many more checks are performed in `add_node`, that can't be done here without duplicating that method.

I propose we extract the validation logic from `add_node` into a method that can be reused, rather than adding this check here. In the meantime I'll make clearer in the docstring and logs that this method performs only JSON schema validation for now.